pkg/binlog/reader/tcp_test.go

```diff
@@ -90,10 +90,23 @@ func (t *testTCPReaderSuite) setUpData(c *C) {
 	query := fmt.Sprintf("DROP DATABASE `%s`", dbName)
 	_, err := t.db.Exec(query)
 
-	// delete previous binlog files/events.
-	query = "RESET MASTER"
-	_, err = t.db.Exec(query)
-	c.Assert(err, IsNil)
+	backoff := 5
+	waitTime := 5 * time.Second
+	waitFn := func() bool {
+		// delete previous binlog files/events. if other test cases writing events, they may be failed.
+		query = "RESET MASTER"
+		_, err = t.db.Exec(query)
+		c.Assert(err, IsNil)
+		// check whether other test cases have wrote any events.
+		time.Sleep(time.Second)
+		_, gs, err2 := utils.GetMasterStatus(t.db, flavor)
+		c.Assert(err2, IsNil)
+		if len(gs.String()) > 0 {
+			return false // some events exist now, try again later.
+		}
+		return true
+	}
+	utils.WaitSomething(backoff, waitTime, waitFn)
```

Comment:
`WaitSomething` can only decrease the possibility of errors.
I observed after `RESET MASTER`, other test cases write binlog event, may it because multi cases run in different goroutine at the same time?

Reply:
yes?

Reply:
1. I observer the fact
2. see <https://stackoverflow.com/questions/24375966/does-go-test-run-unit-tests-concurrently>.